lib/data/network/interceptor/dynamic_url_interceptor.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:dio/dio.dart';
+
+class DynamicUrlInterceptors extends InterceptorsWrapper {
+  String? _baseUrl;
+
+  void changeBaseUrl(String? url) {
+    _baseUrl = url;
+  }
+
+  String? get baseUrl => _baseUrl;
+
```

Comment:
IMO, use String? baseUrl for simplicity.

Reply:
👍